server/webhookNotifier.ts

```diff
@@ -97,58 +112,138 @@ const send = async (models, content: WebhookContent) => {
       chainOrCommwebhookUrls.push(wh.url);
     }
   });
+
+  const {
+    community, actor, action, actedOn, actedOnLink, notificationTitlePrefix, notificationExcerpt, notificationPreviewImageUrl, // forum events
+    title, chainEventLink, fulltext // chain events
+  } = getFilteredContent(content, address);
+  const isChainEvent = !!chainEventLink;
+
+  let actorAvatarUrl = null;
+  const actorAccountLink = address ? `${SERVER_URL}/${address.chain}/account/${address.address}` : null;
+
+  if (address?.id) {
+    const actorProfile = await models.OffchainProfile.findOne({ where: { address_id: address.id } });
+    if (actorProfile?.data) {
+      actorAvatarUrl = JSON.parse(actorProfile.data).avatarUrl;
+    }
+  }
+
+  let previewImageUrl = null; // image url of webhook preview
+  let previewAltText = null; // Alt text of preview image
+
+  // First case
+  if (!isChainEvent) {
+    // if offchain event (thread or comment), need to show embedded image as preview
+    if (notificationPreviewImageUrl) {
+      previewImageUrl = notificationPreviewImageUrl;
+      previewAltText = 'Embedded';
+    }
+  }
+
+  // Second case
+  if (!previewImageUrl) {
+    if (content.chain) {
+      // if the chain has a logo, show it as preview image
+      const chain = await models.Chain.findOne({ where: { id: content.chain } });
+      if (chain) {
+        previewImageUrl = `https://commonwealth.im${chain.icon_url}`;
+        // can't handle the prefix of `previeImageUrl` with SERVER_URL
+        // because social platforms can't access to localhost:8080.
+        previewAltText = chain.name;
+      }
+    } else if (content.community) {
+      // TODO:
+      // if the community has a logo, show it as preview image
```

Comment:
What do you need to complete this TODO?

Reply:
Thanks for reminding me. I've added missing part